backend/graminsta/core/services.py

```diff
@@ -33,3 +35,34 @@ def create_userinfo(validated_data):
         gender=validated_data.pop('gender')
     )
     return user_info
+
+
+def user_authentication(request):
+    """User Authentication
+
+    Parameters
+    ----------
+    request: rest_framework.request.Request
+        Request from the frontend
+
+    Returns
+    -------
+    token.key: String format
+        if the authentication passed.
+    None if the authentication failed.
+    """
+    data = request.data
+    user = authenticate(
+        request,
+        username=data.get("username", ""),
+        password=data.get("password", ""))
+    if user is not None:
+        login(request, user)
+        try:
+            token = Token.objects.get(user=user)
+            token.delete()
+            token = Token.objects.create(user=user)
+        except Token.DoesNotExist:
+            token = Token.objects.create(user=user)
```

Comment:
I think this can be better?

```
try:
  Token.objects.get(user=user).delete()
except Token.DoesNotExist:
  # It's OK if there is no existing token.
   pass
finally:
   return Token.objects.create(user=user)
```